pkg/scheduler/event_handler.go

```diff
@@ -245,10 +245,13 @@ func (s *Scheduler) enqueueAffectedPolicy(newCluster *clusterv1alpha1.Cluster) {
 		affinity := policy.Spec.Placement.ClusterAffinity
 		switch {
 		case affinity == nil:
-			// If no clusters specified, add it in queue
+			// If no clusters specified, add it to the queue
 			fallthrough
 		case util.ClusterMatches(newCluster, *affinity):
-			// If specific cluster matches the affinity. add it in queue
+			// If the new cluster manifest match the affinity, add it to the queue, trigger rescheduling
+			fallthrough
+		case util.ClusterMatches(oldCluster, *affinity):
```

Comment:
I guess it's could be better to merge old and new cluster match.

```go
case util.ClusterMatches(oldCluster, *affinity) != util.ClusterMatches(newCluster, *affinity)
```

Am I right?

Reply:
I think it may not be right. It only triggers rescheduling when the affinity changes, but not when the taints are modified.